master/docs/manual/credentialmanagement.rst

```diff
@@ -0,0 +1,100 @@
+Credential Management
+=====================
+
+Definition
+----------
+
+Buildbot needs to send and/or execute commands requiring credentials.
+
+The goal is to store credentials used by Buildbot commands.
```

Comment:
@tardyp I don't think this document is not intended to be included in the docs as-is, but is rather aimed at the developers to get feedback on the idea. Thus responding to specific details of the wording, formatting and organization in the documentation probably isn't relevant at this point.